app/main.py

```diff
@@ -1,13 +1,40 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error.get("line_number"),
+            "column": error.get("column_number"),
+            "message": error.get("text"),
+            "name": error.get("code"),
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return ({
+        "errors": [
+            {
+                "line": error.get("line_number"),
+                "column": error.get("column_number"),
+                "message": error.get("text"),
+                "name": error.get("code"),
+                "source": "flake8"
+            } for error in errors
+        ],
+        "path": file_path,
+        "status": "passed" if not errors else "failed"
+    } if file_path else {})
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {
+                    "line": error.get("line_number"),
+                    "column": error.get("column_number"),
+                    "message": error.get("text"),
+                    "name": error.get("code"),
+                    "source": "flake8"
+                } for error in linter_report[file_path]
+            ],
+            "path": file_path,
+            "status": "passed" if not linter_report[file_path] else "failed"
+        } for file_path in linter_report
+    ]
```

Comment:
Follow DRY approach and call functions within other functions to avoid code duplication